### PR TITLE
[python] Move output formatter to request level

### PR DESCRIPTION
**Pull request by xyang16**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

**Comment by siddvenk** [comment]
Some of the logic in parse_input is duplicated across handlers, but I think that is ok for now. I'm looking into consolidating this method into a single one across all handlers, so i can try to refactor that separately.

**Review by siddvenk — APPROVED** [review]

File: engines/python/setup/djl_python/deepspeed.py [hunk 1]
```diff
@@ -94,6 +94,11 @@
     "text2text-generation": AutoModelForSeq2SeqLM
 }
 
+OUTPUT_FORMATTER_TO_CONTENT_TYPE = {
```
Comment (frankfliu) [inline engines/python/setup/djl_python/deepspeed.py:97]: Can we move this constant to rolling_batch.py?

File: engines/python/setup/djl_python/huggingface.py [hunk 2]
```diff
@@ -308,12 +313,10 @@ def inference(self, inputs):
...
-                if formatter == "json":
-                    outputs.add_property(f"batch_{i}_Content-Type",
-                                         "application/json")
-                elif formatter == "jsonlines":
-                    outputs.add_property(f"batch_{i}_Content-Type",
-                                         "application/jsonlines")
+                if formatter in OUTPUT_FORMATTER_TO_CONTENT_TYPE:
+                    outputs.add_property(
+                        f"batch_{i}_Content-Type",
+                        OUTPUT_FORMATTER_TO_CONTENT_TYPE[formatter])
```
Comment (frankfliu) [inline engines/python/setup/djl_python/huggingface.py:319]: The formatter might not be a str type, better use a function

File: engines/python/setup/djl_python/deepspeed.py (hunk 1 above)
Comment (xyang16) [reply engines/python/setup/djl_python/deepspeed.py:97]: Done

File: engines/python/setup/djl_python/huggingface.py (hunk 2 above)
Comment (xyang16) [reply engines/python/setup/djl_python/huggingface.py]: Done

File: engines/python/setup/djl_python/deepspeed.py [hunk 3]
```diff
@@ -128,8 +128,6 @@ def initialize(self, properties: dict):
                 "max_seq_len": int(properties.get("max_tokens", 1024)),
                 "tokenizer": self.tokenizer
             }
-            if "output_formatter" in properties:
```
Comment (frankfliu) [inline engines/python/setup/djl_python/deepspeed.py:132]: Why we remove this? Should the value in `serving.properties` be the default value?

File: engines/python/setup/djl_python/deepspeed.py [hunk 4]
```diff
@@ -441,9 +442,13 @@ def inference(self, inputs: Input):
                                 batch_index=i)
                     idx += 1
 
-            content_type = self.rolling_batch.get_content_type()
-            if content_type:
-                outputs.add_property("content-type", content_type)
+                formatter = parameters[i].get("output_formatter")
+                content_type = get_content_type_from_output_formatter(
```
Comment (siddvenk) [inline engines/python/setup/djl_python/deepspeed.py:446]: could we make content-type a property of the request object that we determine based on the output_formatter passed in, and then return it as part of the result from the inference call?

File: engines/python/setup/djl_python/deepspeed.py (hunk 3 above)
Comment (xyang16) [reply engines/python/setup/djl_python/deepspeed.py:132]: This is setting output_formatter in kwargs to initialize DeepSpeedRollingBatch, but we moved output_formatter out of DeepSpeedRollingBatch to request level. So it's not needed.

File: engines/python/setup/djl_python/deepspeed.py (hunk 4 above)
Comment (xyang16) [reply engines/python/setup/djl_python/deepspeed.py:446]: Yes it's already returned in the response in the property.

**Review by frankfliu — APPROVED** [review]

File: engines/python/setup/djl_python/transformers_neuronx.py [hunk 5]
```diff
@@ -130,6 +128,11 @@ def parse_input(self, inputs):
                         raise ValueError(
                             "In order to enable dynamic batching, all input batches must have the same parameters"
                         )
+
+                if not "output_formatter" in param:
+                    param[
+                        "output_formatter"] = self.properties.output_formatter
```
Comment (a-ys) [inline engines/python/setup/djl_python/transformers_neuronx.py:134]: Hi @xyang16, self.properties is not defined for TransformersNeuronXService here. Apologies if this is already planned and I am missing context.

**Pull request merged**